### ISSUE TITLE
When I fetch my model file using query builder, the search part does not work

### ISSUE THREAD

**Issue by MuhammedVeysiGuler** [yajra/laravel-datatables#2967]
I'm using => php 7.4, laravel 6.2 and yajra-datatable-oracle 9.2

The datatable.php file under App/Config is the standard file, I didn't make any changes

if i use Model::get() method i get run time exception error because it works too slow, when i use Model::query() it speeds up but search part doesn't work.

When i use Model::query() When I type in the search part, it becomes success very quickly and the first 10 values ​​in the default table are constantly coming.

My Ajax Controller
```
 // $basvurular = Basvuru::get();   // too slow and search box working
            $basvurular = Basvuru::query();   // fast but search box not working and i wanna use this
            return DataTables::of($basvurular)
                ->addColumn('aday_no', function ($data) {
                    return "<td>$data->id</td>";
                })
                ->addColumn('aday_foto', function ($data) {
                    return '<td class="text-center">
                                <img class="img-profile rounded-circle" src="'.url($data->kullanici->fotograf).'"
                                     style="width: 64px; height: 64px">
                                <br><a href="'.url($data->kullanici->fotograf).'"
                                       target="_blank">'.__('genel.indir').'</a>
                            </td>';
                })
                ->addColumn('basvuru_donem', function ($data) {
                    return "<td>".$data->basvuru_donem->adi_tr."</td>";
                })
                ->addColumn('aday_isim_soyisim', function ($data) {
                    return "<td>".$data->kullanici->name . '&nbsp;<br>' . $data->kullanici->surname."</td>";
                })
                ->addColumn('aday_mail', function ($data) {
                    return "<td>".str_replace("@","<br>@",$data->kullanici->email)."</td>";
                })
                ->addColumn('aday_telefon', function ($data) {
                    return "<td>".$data->telefon."</td>";
                })
                ->addColumn('aday_uyruk', function ($data) {
                    return "<td>".$data->birinci_uyruk->adi_tr."</td>";
                })
                ->addColumn('aday_uyruk2', function ($data) {
                    if(isset($data->ikinci_uyruk)){
                        return "<td>".$data->ikinci_uyruk->adi_tr."</td>";
                    }
                    return "<td> </td>";
                })
                ->addColumn('sinav_yeri', function ($data) {
                    if($data->disardan_mi == 0){
                        return "<td>".$data->sinav_yeri->adi_tr."</td>";
                    }
                    return "<td>". __('basvuru.kayit_disardan') ."</td>";
                })
                ->addColumn('odeme', function ($data) {
                    if($data->odemis_mi && $data->disardan_mi == 0)
                        return  "<td>". __('basvuru.basvuru_durum_evet', ['tutar'=> $data->banka->amount])."</td>";
                    elseif($data->disardan_mi == 1){
                        return "<td>". __('basvuru.kayit_disardan')."</td>";
                    }else{
                        return "<td>
                                    <button class=\"btn btn-sm btn-primary\"
                                            onclick=\"location.href='".route('y_banka_redirect',$data->id)."'\">"
                            . __('basvuru.basvuru_durum_hayir')."
                                        </button>
                                    </td>";
                    }

                })
                ->addColumn('basvuru_durumu', function ($data) {
                    return "<td>".$data->basvuru_durum->adi_tr."</td>";
                })
                ->addColumn('basvuru_aciklama', function ($data) {
                    return "<td>".$data->aciklama."</td>";
                })
                ->addColumn('basvuru_egitim', function ($data) {
                    if ($data->egitim_verildi_mi){
                        return "<td>".__('genel.evet')."</td>";
                    }
                    return "<td>".__('genel.hayir')."</td>";
                })
                ->addColumn('salon_yeri', function ($data) {
                    if ($data->disardan_mi == 0){
                        if(isset($data->salon_islemleri)){
                            return "<td>
                                        <b>Adres: </b>". $data->salon_islemleri->bina_islemleri->adres."<br>
                                        <b>Salon: </b>". $data->salon_islemleri->salon_ad.
                                    "</td>";
                        }
                    }else{
                        return "<td>".__('basvuru.kayit_disardan')."</td>";
                    }
                })
                ->addColumn('kayit_tarihi', function ($data) {
                    return "<td>". (new DateTime($data->created_at))->format('d/m/Y H:i:s') ."</td>";
                })
                ->addColumn('islemler', function ($data) {
                    return "<td><button class=\"btn btn-sm btn-primary\"
                                        onclick=\"location.href='".route('y_basvuru_edit', $data->id)."'\">".
                                    __('basvuru.islemler_guncelle')."
                                </button>
                                <button id=\"sablon__sil_modal_".$data->id."\"
                                        class=\"btn btn-sm btn-danger sablon__sil_modal\" data-toggle=\"modal\"
                                        data-target=\"#sablon__sil_modal\">". __('genel.sil') ."
                    </button></td>";
                })
                ->rawColumns(['aday_no','aday_foto','basvuru_donem','aday_isim_soyisim','aday_mail','aday_telefon','aday_uyruk','aday_uyruk2','sinav_yeri','odeme','basvuru_durumu','basvuru_aciklama','basvuru_egitim','salon_yeri','kayit_tarihi','islemler'])
                ->make(true);
```

My Model
```
class Basvuru extends Model
{
    protected $table = 'basvuru';
    protected $guarded = ['id'];

    //relations
    public function basvuru_donem()
    {
        return $this->belongsTo(BasvuruDonem::class, 'basvuru_donem_id', 'id');
    }

    public function kullanici()
    {
        return $this->belongsTo(User::class, 'kullanici_id', 'id');
    }

    public function cinsiyet()
    {
        return $this->belongsTo(Cinsiyet::class, 'cinsiyet_id', 'id');
    }

    public function birinci_uyruk()
    {
        return $this->belongsTo(Ulke::class, 'birinci_uyruk_id', 'id');
    }

    public function ikinci_uyruk()
    {
        return $this->belongsTo(Ulke::class, 'ikinci_uyruk_id', 'id');
    }

    public function lise_ulke()
    {
        return $this->belongsTo(Ulke::class, 'lise_ulke_id', 'id');
    }

    public function sinav_yeri()
    {
        return $this->belongsTo(SinavMerkezi::class, 'sinav_yeri_id', 'id');
    }

    public function banka()
    {
        return $this->belongsTo(Banka::class, 'banka_id', 'id');
    }

    public function basvuru_durum()
    {
        return $this->belongsTo(BasvuruDurum::class, 'basvuru_durum_id', 'id');
    }

    public function salon_islemleri()
    {
        return $this->belongsTo(Salonİslemleri::class, 'salon_islemleri_id', 'id');
    }

    public function tercih_1_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih1_id', 'id');
    }

    public function tercih_2_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih2_id', 'id');
    }

    public function tercih_3_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih3_id', 'id');
    }

    public function tercih_4_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih4_id', 'id');
    }

    public function tercih_5_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih5_id', 'id');
    }

    public function tercih_6_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih6_id', 'id');
    }

    public function tercih_7_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih7_id', 'id');
    }

    public function tercih_8_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih8_id', 'id');
    }

    public function tercih_9_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih9_id', 'id');
    }

    public function tercih_10_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'tercih10_id', 'id');
    }

    public function yerlestirme_bolum()
    {
        return $this->belongsTo(TercihBolum::class, 'yerlestirme_id', 'id');
    }

    public function kayit_durum()
    {
        return $this->belongsTo(BasvuruDurum::class, 'kayit_durum_id', 'id');
    }
```
My Ajax Blade
```
<div class="card">
            <div class="card-body">
                <table id="basvuru_tablo2" class="table table-striped table-bordered dt-responsive" style="width:100%">
                </table>
            </div>
        </div>


dataTable = $('#basvuru_tablo2').DataTable({
            language: {
                url: '//cdn.datatables.net/plug-ins/1.10.21/i18n/Turkish.json'
            },
            order: [
                [0, 'ASC']
            ],
            processing: true,
            serverSide: true,
            scrollX: true,
            scrollY: true,
            ajax: '{!! route('y_basvuru_index_table') !!}',
            columns: [
                {data: 'aday_no', title:'{{ __('basvuru.aday_no') }}'},
                {data: 'aday_foto', title:'{{ __('basvuru.fotograf') }}'},
                {data: 'basvuru_donem', title:'{{ __('basvuru.basvuru_donem') }}'},
                {data: 'aday_isim_soyisim', title:'{{ __('basvuru.isim') . ' ' . __('basvuru.soyisim') }}'},
                {data: 'aday_mail', title:'{{ __('basvuru.email') }}'},
                {data: 'aday_telefon', title:'{{ __('basvuru.telefon') }}'},
                {data: 'aday_uyruk', title:'{{ __('basvuru.birinci_uyruk') }}'},
                {data: 'aday_uyruk2', title:'{{ __('basvuru.ikinci_uyruk') }}'},
                {data: 'sinav_yeri', title:'{{ __('basvuru.sinav_yeri') }}'},
                {data: 'odeme', title:'{{ __('basvuru.banka_durum') }}'},
                {data: 'basvuru_durumu', title:'{{ __('basvuru.basvuru_durum') }}'},
                {data: 'basvuru_aciklama', title:'{{ __('basvuru.basvuru_aciklama') }}'},
                {data: 'basvuru_egitim', title:'{{ __('basvuru.egitim_verildi_mi') }}'},
                {data: 'salon_yeri', title:'{{ __('basvuru.salon_yeri') }}'},
                {data: 'kayit_tarihi', title:'{{ __('basvuru.kayit_tarihi') }}'},
                {data: 'islemler', title:'{{ __('basvuru.islemler') }}'},
            ],
            dom: "Blfrtip",
            buttons: [
                {
                    text: 'Kolonlar',
                    extend: 'colvis',
                },
                {
                    text: 'Csv',
                    extend: 'csvHtml5',
                },
                {
                    text: 'Excel',
                    extend: 'excelHtml5',
                },
                {
                    text: 'PDF',
                    extend: 'pdfHtml5',
                },
                {
                    text: 'Print',
                    extend: 'print',
                },
            ],
            columnDefs: [{
                orderable: true,
                targets: -1
            }],

        });
```
To summarize: When I do the get function, the default search part of the datatable works without any problems, but when I do the query with the query function, the default search part of the datatable does not work and as a result, the first 10 data is always returned. It doesn't mean anything if I write an empty input or a random word bc not working why ??
